crowdfunding/projects/views.py

```diff
@@ -1,3 +1,124 @@
+from django_filters.rest_framework import DjangoFilterBackend
+from django.http import Http404
 from django.shortcuts import render
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status, permissions, generics
+from .models import Project, Pledge, ProjectUpdates, LikedBy
+from .serializers import ProjectSerializer, PledgeSerializer, ProjectDetailSerializer, ProjectUpdatesSerializer, LikedBySerializer
+from .permissions import IsOwnerOrReadOnly
 
 # Create your views here.
+# class ProjectList(generics.ListCreateAPIView):
+class ProjectList(APIView):
+    permission_classes = [permissions.IsAuthenticatedOrReadOnly]
+
+    search_fields = ['title', 'description']
```

Comment:
I'm not sure this will work because you aren't applying to filters in your get method